mapeditor/mapcontroller.cpp

```diff
@@ -323,6 +324,48 @@ void MapController::commitObjectErase(int level)
 	main->mapChanged();
 }
 
+void MapController::copyToClipboard(int level)
+{
+	_clipboard.clear();
+	_clipboardShiftIndex = 0;
+	auto selectedObjects = _scenes[level]->selectionObjectsView.getSelection();
+	for(auto * obj : selectedObjects)
+	{
+		assert(obj->pos.z == level);
+		_clipboard.push_back(CMemorySerializer::deepCopy(*obj));
+	}
+}
+
+void MapController::pasteFromClipboard(int level)
+{
+	_scenes[level]->selectionObjectsView.clear();
+	
+	auto shift = int3::getDirs()[_clipboardShiftIndex++];
+	if(_clipboardShiftIndex == int3::getDirs().size())
+		_clipboardShiftIndex = 0;
+	
+	for(auto & objuptr : _clipboard)
```

Comment:
`objuptr` is a strange name for a variable. We also usually use camelCase, so should be `objPtr`. And what "u" in variable name stands for?

Reply:
Unique ptr

Reply:
objuptr -> objUniquePtr

Reply:
Is that fact important here? You don't do anything specifically different on basis that this is unique ptr.
If you were to change unique_ptr to shared_ptr this loop won't change. So no need to specify this here.
And even if decide to keep it - then what @dydzio0614 said.